<launch>

    <!-- setting simulation-->
    <param name="/use_sim_time" value="true"/>

    <!-- Static transform from base_link to rslidar -->
    <node pkg="tf2_ros" type="static_transform_publisher" name="static_tf_pub_rslidar" 
          args="0 0 0.245 0 0 0 1 base_link rslidar" /> 

    <!-- Static transform from base_link to laser_frame -->
    <node pkg="tf2_ros" type="static_transform_publisher" name="static_tf_pub_laser_frame" 
          args="0 0 0 0 0 0 1 base_link laser_frame" />

    <!-- Node that converts point cloud to laser scan -->
    <node pkg="pointcloud_to_laserscan" type="pointcloud_to_laserscan_node" name="pointcloud_to_laserscan" output="screen">
        <!-- Topic remapping -->
        <remap from="cloud_in" to="/ugv/rslidar_points" />
        <param name="target_frame" value="laser_frame"/>
        <param name="transform_tolerance" value="0.1"/>
        <param name="min_height" value="0.0"/>
        <param name="max_height" value="1.0"/>
        <param name="angle_min" value="-3.14159265359"/>            <!-- Minimum angle -->
        <param name="angle_max" value="3.14159265359"/>             <!-- Maximum angle -->
        <param name="angle_increment" value="0.00872665"/>          <!-- Angle increment -->
        <param name="scan_time" value="0.1"/>
        <param name="range_min" value="0.45"/>
        <param name="range_max" value="10.0"/>
        <param name="use_inf" value="true"/>
        <param name="inf_epsilon" value="1.0"/>
        <remap from="scan" to="/scan" /> <!-- Ensure this matches with the scan topic in Gmapping -->
    </node>
    
    <!-- Node to publish odom on tf-->
    <node pkg="second_project" type="ugv_odom_to_tf" name="ugv_odom_to_tf" output="screen" />

    <!-- slam toolbox -->
    <include file="$(find second_project)/launch/slam_toolbox_default.launch"/>


    <!-- Starting RViz-->
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find second_project)/rviz/robot_mapping.rviz" />


</launch>
